google/cloud/aiplatform/metadata/schema/system_schema.py

```diff
@@ -0,0 +1,508 @@
+
+    def __init__(
+        self,
+        model_name: Optional[str] = None,
+        uri: Optional[str] = None,
+        display_name: Optional[str] = None,
+        schema_version: Optional[str] = None,
+        description: Optional[str] = None,
+        metadata: Optional[Dict] = None,
+        state: gca_artifact.Artifact.State = gca_artifact.Artifact.State.LIVE,
+        **kwargs,
+    ):
+        """Args:
+        model_name (str):
+            Optional. The resource name of the Artifact following the format as follows.
+            This is globally unique in a metadataStore:
+            projects/123/locations/us-central1/metadataStores/<metadata_store_id>/artifacts/<resource_id>.
+        uri (str):
+            Optional. The uniform resource identifier of the artifact file. May be empty if there is no actual
+            artifact file.
+        display_name (str):
+            Optional. The user-defined name of the base.
+        schema_version (str):
+            Optional. schema_version specifies the version used by the base.
+            If not set, defaults to use the latest version.
+        description (str):
+            Optional. Describes the purpose of the Artifact to be created.
+        metadata (Dict):
+            Optional. Contains the metadata information that will be stored in the Artifact.
+        state (google.cloud.gapic.types.Artifact.State):
+            Optional. The state of this Artifact. This is a
+            property of the Artifact, and does not imply or
+            capture any ongoing process. This property is
+            managed by clients (such as Vertex AI
+            Pipelines), and the system does not prescribe or
+            check the validity of state transitions.
+        **kwargs:
+            Optional. Additional Args that will be passed directly to the Artifact base method for backward compatibility.
+        """
+        extended_metadata = metadata or {}
+        extended_metadata["resourceName"] = model_name
+        super(Model, self).__init__(
+            schema_title=self.SCHEMA_TITLE,
+            resource_name=model_name,
+            uri=uri,
+            display_name=display_name,
+            schema_version=schema_version,
+            description=description,
+            metadata=extended_metadata,
+            state=state,
+            kwargs=kwargs,
+        )
+
+
+class Dataset(base_artifact.BaseArtifactSchema):
+    """An artifact representing a system Dataset."""
+
+    SCHEMA_TITLE = "system.Dataset"
+
+    def __init__(
+        self,
+        dataset_name: Optional[str] = None,
+        uri: Optional[str] = None,
+        display_name: Optional[str] = None,
+        schema_version: Optional[str] = None,
+        description: Optional[str] = None,
+        metadata: Optional[Dict] = None,
+        state: gca_artifact.Artifact.State = gca_artifact.Artifact.State.LIVE,
+        **kwargs,
+    ):
+        """Args:
+        dataset_name (str):
```

Comment:
Same question. Also, why is this an artifact resource name?

Reply:
Done. That is a mistake system types should not have a resouce_name. removing across the board.